prefect_gcp/workers/vertex.py

```diff
@@ -0,0 +1,633 @@
+            "See https://cloud.google.com/vertex-ai/docs/reference/rest/v1/MachineSpec"
+        ),
+        example="NVIDIA_TESLA_K80",
+    )
+    accelerator_count: Optional[int] = Field(
+        title="Accelerator Count",
+        description=(
+            "The number of accelerators to attach to the machine. "
+            "See https://cloud.google.com/vertex-ai/docs/reference/rest/v1/MachineSpec"
+        ),
+    )
+    boot_disk_type: str = Field(
+        title="Boot Disk Type",
+        description="The type of boot disk to attach to the machine.",
+        example="pd-ssd",
+    )
+    boot_disk_size_gb: int = Field(
+        title="Boot Disk Size (GB)",
+        description="The size of the boot disk to attach to the machine, in gigabytes.",
+        example=100,
+    )
+    maximum_run_time_hours: int = Field(
+        default=1,
+        title="Maximum Run Time (Hours)",
+        description="The maximum job running time, in hours",
+    )
+    network: Optional[str] = Field(
+        default=None,
+        title="Network",
+        description="The full name of the Compute Engine network"
+        "to which the Job should be peered. Private services access must "
+        "already be configured for the network. If left unspecified, the job "
+        "is not peered with any network. "
+        "For example: projects/12345/global/networks/myVPC",
+    )
+    reserved_ip_ranges: Optional[List[str]] = Field(
+        default=None,
+        title="Reserved IP Ranges",
+        description="A list of names for the reserved ip ranges under the VPC "
+        "network that can be used for this job. If set, we will deploy the job "
+        "within the provided ip ranges. Otherwise, the job will be deployed to "
+        "any ip ranges under the provided VPC network.",
+    )
+    service_account_name: Optional[str] = Field(
+        default=None,
+        title="Service Account Name",
+        description=(
+            "Specifies the service account to use "
+            "as the run-as account in Vertex AI. The worker submitting jobs must have "
+            "act-as permission on this run-as account. If unspecified, the AI "
+            "Platform Custom Code Service Agent for the CustomJob's project is "
+            "used. Takes precedence over the service account found in GCP credentials, "
+            "and required if a service account cannot be detected in GCP credentials."
+        ),
+    )
+    job_watch_poll_interval: float = Field(
+        default=5.0,
+        title="Poll Interval (Seconds)",
+        description=(
+            "The amount of time to wait between GCP API calls while monitoring the "
+            "state of a Vertex AI Job."
+        ),
+    )
+
+
+def _get_base_job_spec() -> Dict[str, Any]:
+    """Returns a base job body to use for job spec validation.
+    Note that the values are stubbed and are not used for the actual job."""
+    return {
+        "maximum_run_time_hours": "1",
```

Comment:
im just validating against the presence of keys, so not going to validate against the provided values (since they generally come from a set of acceptable values)